app/lib/actions/katello/host/update.rb

```diff
@@ -16,23 +14,35 @@ def plan(host, consumer_params = nil)
                 consumer_params = host.subscription_facet.consumer_attributes
               end
 
-              host.subscription_facet.update_from_consumer_attributes(consumer_params)
-              host.subscription_facet.save!
-              plan_action(::Actions::Candlepin::Consumer::Update, host.subscription_facet.uuid, consumer_params)
+              cp_update = plan_action(::Actions::Candlepin::Consumer::Update, host.subscription_facet.uuid, consumer_params)
+              cp_updated = cp_update.output[:response]
             end
 
-            if host.subscription_facet.try(:autoheal)
+            if consumer_params.present? && consumer_params['autoheal']
               plan_action(::Actions::Candlepin::Consumer::AutoAttachSubscriptions, :uuid => host.subscription_facet.uuid)
             end
+
+            plan_self(:hostname => host.name, :consumer_params => consumer_params, :host_id => host.id,
+                      :cp_updated => cp_updated)
```

Comment:
I don't see cp_updated used anywhere aside from passing it in?,   I don't particularly like the name of that variable either :)